test/transitive_dependency/CMakeLists.txt

```diff
@@ -30,6 +30,15 @@ add_library(${PROJECT_NAME} SHARED source/transitive_dependency.cpp)
 target_compile_features(${PROJECT_NAME} PRIVATE cxx_std_17)
 target_link_libraries(${PROJECT_NAME} PUBLIC fmt::fmt-header-only cxxopts)
 
+if(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.23.0")
+  # required in CMake 3.23 and more
+  set(CMAKE_VERIFY_INTERFACE_HEADER_SETS ON)
+  target_sources(
+    ${PROJECT_NAME} PUBLIC FILE_SET HEADERS BASE_DIRS include FILES
+                           include/transitive_dependency/transitive_dependency.h
```

Comment:
Now, this header is included in a `CMake generated target` and may be checked and compiled.